src/Apps/Artist/Routes/Consign/Components/ArtistConsignHeader.tsx

```diff
@@ -46,3 +68,137 @@ export const ArtistConsignHeader: React.FC<ArtistConsignHeaderProps> = props =>
     </SectionContainer>
   )
 }
+
+/**
+ * Displays a random set of images for a given consignment artist.
+ *
+ * Some notes about implementation:
+ *
+ * 1) Due to responsiveness and SSR we need to dynamically scale the width and the
+ *    height of the image based on overall width of the container.
+ * 2) For this we use our `<ResponsiveImage>` component.
+ * 3) However, this component achieves responsivity by forcing padding within a
+ *    container and then applying a "fit" to a background image.
+ * 4) Therefore, conventional borders cannot be applied, and a CSS filter is
+ *    required. This is based upon pixel data, and border faked.
+ * 5) The pixel data comes from a second image, sitting below the first, with
+ *    its brightness turned to -100.
+ * 5) It's not perfect, and the layout needs to also account for variable
+ *    aspect ratios.
+ * 6) The positioning CSS is also tricky, so please forgive...
+ */
+const HeaderImages = (props: {
+  artworksByInternalID: Consign_artworksByInternalID
+}) => {
+  const getRandomImages = () => {
+    const [left, right] = sampleSize(props.artworksByInternalID, 2)
+
+    // Layout requires that we only use vertically oriented images
+    const widthExceedsHeight = ({ image }) => {
+      return image.resized.width > image.resized.height
+    }
+    if (widthExceedsHeight(left)) {
+      return getRandomImages()
+    }
+    if (widthExceedsHeight(right)) {
+      return getRandomImages()
+    }
+
+    return [left, right]
+  }
+
+  const [leftImage, rightImage] = getRandomImages()
+
+  return (
+    <HeaderImageContainer>
```

Comment:
I think it would be cool to extract and name some concepts here. What about something like this:

```ts
const HeaderImages = props => {
  // .. stuff ..
  return (
    <HeaderImageContainer>
      <Flex width="100%" justifyContent="space-between">
        <LeftImage image={leftImage} />
        <RightImage image={leftImage} />
      </Flex>
    </HeaderImageContainer>
  )
}

const LeftImage = props => {
  // .. stuff ..
  return (
    <LeftImageWrapper>
      <BorderImage image={image} />
      <ShrunkImage image={image} />
    </LeftImageWrapper>
  )
}

const RightImage = props => {
  // .. stuff ..
  return (
    <RightImageWrapper>
      <BorderImage image={image} />
      <ShrunkImage image={image} />
    </RightImageWrapper>
  )
}

const BorderImage = props => {
  // .. stuff ..
  return (
    <ResponsiveImage
      src={src}
      width={width}
      style={{
        transformOrigin: "center",
        filter: "invert(100%) brightness(-100%)",
        backgroundPosition: "left",
      }}
    />
  )
}

const ShrunkImage = props => {
  // .. stuff ..
  return (
    <ResponsiveImage
      src={src}
      width={width}
      style={{
        position: "absolute",
        top: 0,
        left: -4,
        transformOrigin: "center",
        backgroundPosition: "left",
        transform: "scaleX(.97) scaleY(.96)",
      }}
    />
  )
}
```

Hopefully I copy/pasted that mostly correct! 😝 So yeah, my approach here is to pull out things and give them names so that I can see the layers and how they all compose together. I think something like this would make this code easier to change. What do you think?

Reply:
Great suggestion 👍 Didn't follow _exactly_, but kept the spirit of it. Much easier to read now.